src/slot.rs

```diff
@@ -0,0 +1,105 @@
+use chrono::{NaiveDateTime, Timelike};
+use serde::{Deserialize, Serialize};
+use std::ops::{Add, Sub};
+
+#[derive(Debug, Eq, PartialEq, Clone, Copy, Deserialize)]
+pub struct Slot {
+    pub start: NaiveDateTime,
+    pub end: NaiveDateTime,
+}
+
+impl Sub for Slot {
+    type Output = Vec<Slot>;
+
+    fn sub(self, other: Self) -> Self::Output {
+        let mut result = Vec::new();
+        if (other.start < self.start) && (other.end <= self.start) {
```

Comment:
I would expect six different cases:
- entirely before
- entirely after
- overlapping begin
- overlapping end
- overlapping begin + end (equal or bigger)
- overlapping within (smaller)

Mind the equals for the slot [begin:end[:
- begin is inclusive
- end is exclusive

Reply:
Subtraction - I count seven if blocks.
Didn't check details.

Reply:
I've just double-checked and I believe all cases are handled. You can verify from the comments in the `if` blocks or from the unit tests from line 377.

Reply:
If we can simplify to 6 if blocks instead of 8 I think we should.